extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -431,72 +438,52 @@ export class AmpStoryReaction extends AMP.BaseElement {
    * @private
    */
   retrieveReactionData_() {
-    return this.executeReactionRequest_(
-      dict({
-        'method': 'GET',
-      })
-    )
-      .then((response) => {
-        this.handleSuccessfulDataRetrieval_(response);
-      })
-      .catch((error) => {
-        dev().error(TAG, error);
-      });
-  }
-
-  /**
-   * Update the Reaction data in the datastore
-   *
-   * @param {number} reactionValue
-   * @private
-   */
-  updateReactionData_(reactionValue) {
-    this.executeReactionRequest_(
-      dict({
-        'method': 'POST',
-      }),
-      reactionValue
-    ).catch((error) => {
-      dev().error(TAG, error);
+    return this.executeReactionRequest_('GET').then((response) => {
+      this.handleSuccessfulDataRetrieval_(response);
     });
   }
 
   /**
    * Executes a Reactions API call.
    *
-   * @param {Object} requestOptions
-   * @param {number=} reactionValue
-   * @return {Promise<ReactionResponseType|undefined>}
+   * @param {string} method GET or POST.
+   * @param {number=} optionSelected
+   * @return {Promise<ReactionResponseType>}
    * @private
    */
-  executeReactionRequest_(requestOptions, reactionValue) {
-    if (!assertAbsoluteHttpOrHttpsUrl(this.element.getAttribute('endpoint'))) {
+  executeReactionRequest_(method, optionSelected) {
+    let url = this.element.getAttribute('endpoint');
+    if (!assertAbsoluteHttpOrHttpsUrl(url)) {
       return Promise.reject(ENDPOINT_INVALID_ERROR);
     }
 
-    if (this.reactionId_ === null) {
+    if (!this.reactionId_) {
       const pageId = closest(dev().assertElement(this.element), (el) => {
         return el.tagName.toLowerCase() === 'amp-story-page';
       }).getAttribute('id');
-
-      this.reactionId_ = `CANONICAL_URL#page=${pageId}`;
+      this.reactionId_ = `CANONICAL_URL+${pageId}`;
+      if (this.element.hasAttribute('id')) {
+        this.reactionId_ += '+' + this.element.getAttribute('id');
+      }
```

Comment:
Element id is a better tracker than page ID in any case, so we definitely should use the element id. Removing the page ID would also help if someone wanted to move a quiz from one page to another, but we need to ensure every reaction element has an ID before going that route. But still, adding the id allows for multiple reactions to live in the same page without errors until we can validate that there must only be one per page (if we want).